add_trailing_comma/_plugins/pep695.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import ast
+import sys
+from typing import Iterable
+
+from tokenize_rt import Offset
+from tokenize_rt import Token
+
+from add_trailing_comma._ast_helpers import ast_to_offset
+from add_trailing_comma._data import register
+from add_trailing_comma._data import State
+from add_trailing_comma._data import TokenFunc
+from add_trailing_comma._token_helpers import find_simple
+from add_trailing_comma._token_helpers import fix_brace
+
+
+if sys.version_info >= (3, 12):  # pragma: >=3.12 cover
+    def _fix_pep695(
+        i: int,
+        tokens: list[Token],
+    ) -> None:
+        # When we get a `FunctionDef`, `ClassDef`, or `TypeAlias` object,
+        # the ast refers to it as this.
+        #     class ClassA[T1, T2, T3](list[T1]):
+        #     ^ where AST points
+        # We really care about the opening bracket though, so instead send
+        # in the offset of the first type-param.
+        #     class ClassA[T1, T2, T3](list[T1]):
+        #                  ^ first type-param
+        # This function will then move towards the start until we find an
+        # opening bracket.
+        #     class ClassA[T1, T2, T3](list[T1]):
+        #                 ^ opening bracket
+        # This works for all of `ClassDef`, `FunctionDef` and `TypeAlias`.
+        for n in range(i, -1, -1):
+            token = tokens[n]
+            if token.name == 'OP' and token.src == '[':
+                return fix_brace(
+                    tokens,
+                    find_simple(n, tokens),
+                    add_comma=True,
+                    remove_comma=True,
+                )
+        else:
+            raise AssertionError('Past beginning?')
+
+    @register(ast.FunctionDef)
+    def vitis_FunctionDef(
+        state: State,
+        node: ast.FunctionDef,
+    ) -> Iterable[tuple[Offset, TokenFunc]]:
+        if node.type_params:
+            yield ast_to_offset(node.type_params[0]), _fix_pep695
```

Comment:
probably can `yield from _whatever_common(...)` -- also the function names are typo'd here